strawberryfields/plot.py

```diff
@@ -120,3 +123,236 @@ def generate_wigner_chart(data, xvec, pvec, contours=True):
+    chart["data"] = [dict() for i in range(numplots)]
+
+    for idx, n in enumerate(sorted(modes)):
+        chart["data"][idx]["type"] = "bar"
+        chart["data"][idx]["marker"] = {"color": "#1f9094"}
+        chart["data"][idx]["x"] = xlabels
+        chart["data"][idx]["y"] = photon_dists[n].tolist()
+
+        if idx == 0:
+            Xax = ("xaxis", "x")
+        else:
+            chart["layout"]["annotations"].append(copy(chart["layout"]["annotations"][idx - 1]))
+            Xax = ("xaxis{}".format(idx + 1), "x{}".format(idx + 1))
+
+        chart["data"][idx]["xaxis"] = Xax[1]
+        chart["data"][idx]["yaxis"] = "y"
+        chart["data"][idx]["name"] = ""
+
+        dXa = 0.01 if idx != 0 else 0
+        dXb = 0.01 if idx != numplots - 1 else 0
+
+        chart["layout"][Xax[0]] = {
+            "type": "category",
+            "domain": [idx / numplots + dXa, (idx + 1) / numplots - dXb],
+            "title": "mode {}".format(n),
+            "fixedrange": True,
+            "gridcolor": "rgba(0,0,0,0)",
+        }
+        val_mean_photon = str(np.round(mean[n], 3))
+        expr_mean_photon = r"$\langle \hat{n} \rangle=" + val_mean_photon + "$"
+        chart["layout"]["annotations"][idx]["text"] = expr_mean_photon
+        chart["layout"]["annotations"][idx]["x"] = idx / numplots + dXa + 0.5 / numplots
+
+    chart["layout"]["xaxis"]["type"] = "category"
+    chart["layout"]["title"] = "Marginal Fock state probabilities"
+
+    chart["layout"]["font"] = {"color": textcolor}
+
+    return chart
+
+
+# Plot.ly default linechart JSON
+linechart_default = {
+    "data": [{"type": "scatter", "x": [], "y": []}],
+    "layout": {
+        "width": 835,
+        "height": 500,
+        "margin": {"l": 100, "r": 100, "b": 100, "t": 100, "pad": 4},
+        "paper_bgcolor": "white",
+        "plot_bgcolor": "white",
+        "xaxis": {
+            "gridcolor": textcolor,
+            "autorange": True,
+            "title": "Quadrature value",
+        },
+        "yaxis": {
+            "gridcolor": textcolor,
+            "autorange": True,
+            "title": "Probability",
+        },
+    },
+    "config": {
+        "modeBarButtonsToRemove": ["lasso2d", "select2d", "toggleSpikelines"],
+        "displaylogo": False,
+    },
+}
+
+
+def plot_quad(state, modes, xvec, pvec, renderer="browser"):
+    """Plot the Wigner function with Plot.ly.
+
+    Args:
+        state (:class:`.BaseState`): the state used for plotting
+        mode (int): mode used to calculate the reduced Wigner function
+        xvec (array): array of discretized :math:`x` quadrature values
+        pvec (array): array of discretized :math:`p` quadrature values
+        renderer (string): the renderer for plotting with Plot.ly
+    """
+    pio = _get_plotly()
+    pio.renderers.default = renderer
+
+    for mode in modes:
+        p_probs = state.p_quad_values(mode, xvec, pvec).tolist()
+        x_probs = state.x_quad_values(mode, xvec, pvec).tolist()
```

Comment:
Perhaps these should also be moved into `generate_quad_chart`?

Reply:
Updated